src/m365/pa/commands/app/app-permission-remove.ts

```diff
@@ -0,0 +1,191 @@
+import { validation } from '../../../../utils/validation';
+import PowerAppsCommand from '../../../base/PowerAppsCommand';
+import commands from '../../commands';
+import { aadUser } from '../../../../utils/aadUser';
+import { accessToken } from '../../../../utils/accessToken';
+import Auth from '../../../../Auth';
+import { Cli } from '../../../../cli/Cli';
+
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  appName: string;
+  userId?: string;
+  userName?: string;
+  groupId?: string;
+  groupName?: string;
+  tenant?: boolean;
+  asAdmin?: boolean;
+  environmentName?: string;
+  confirm?: boolean;
+}
+
+class PaAppPermissionRemoveCommand extends PowerAppsCommand {
+  public get name(): string {
+    return commands.APP_PERMISSION_REMOVE;
+  }
+
+  public get description(): string {
+    return 'Removes permissions to a Power Apps app';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+    this.#initOptionSets();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        userId: typeof args.options.userId !== 'undefined',
+        userName: typeof args.options.userName !== 'undefined',
+        groupId: typeof args.options.userId !== 'undefined',
+        groupName: typeof args.options.userName !== 'undefined',
+        tenant: !!args.options.tenant,
+        asAdmin: !!args.options.asAdmin,
+        environmentName: typeof args.options.environmentName !== 'undefined',
+        confirm: !!args.options.confirm
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '--appName <appName>'
+      },
+      {
+        option: '--userId [userId]'
+      },
+      {
+        option: '--userName [userName]'
+      },
+      {
+        option: '--groupId [groupId]'
+      },
+      {
+        option: '--groupName [groupName]'
+      },
+      {
+        option: '--tenant'
+      },
+      {
+        option: '--asAdmin'
+      },
+      {
+        option: '-e, --environmentName [environmentName]'
+      },
+      {
+        option: '--confirm'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (!validation.isValidGuid(args.options.appName)) {
+          return `${args.options.appName} is not a valid GUID`;
+        }
+
+        if (args.options.userId && !validation.isValidGuid(args.options.userId)) {
+          return `${args.options.userId} is not a valid GUID`;
+        }
+
+        if (args.options.groupId && !validation.isValidGuid(args.options.groupId)) {
+          return `${args.options.groupId} is not a valid GUID`;
+        }
+
+        if (args.options.environmentName && !args.options.asAdmin) {
+          return 'please use asAdmin when using environmentName';
+        }
+
+        return true;
```

Comment:
Let's also check that `environmentName` is filled in and only filled in when specifying `asAdmin`.